File: 1643.py
def CSES1643() -> int:
    n = int(input())
    x = list(map(int, input().split()))
    curr_sum = 0
    max_sum = -float('inf')

    for x_i in x:
        curr_sum = max(x_i, curr_sum + x_i)
        max_sum = max(max_sum, curr_sum)

    return max_sum

print(CSES1643())

File: cses/1643.cpp
#include <bits/stdc++.h>

long long CSES1643() {
    long long n, INF = powl(10, 18);
    std::cin >> n;
    std::vector<long long> a(n);
    for (int i = 0; i < n; i++) {
        std::cin >> a[i];
    }
    long long ans = -INF, curr = 0;
    for (int i = 0; i < n; i++) {
        curr = std::max(a[i], curr + a[i]);
        ans = std::max(ans, curr);
    }
    return ans;
}

int main() {
    std::ios_base::sync_with_stdio(false);
    std::cin.tie(NULL);

    std::cout << CSES1643() << "\n";
    return 0;
}